src/main/java/org/gingesnap/cdc/chain/EventProcessingChainFactory.java

```diff
@@ -0,0 +1,58 @@
+package org.gingesnap.cdc.chain;
+
+import java.util.List;
+import java.util.Optional;
+
+import org.gingesnap.cdc.CacheBackend;
+import org.gingesnap.cdc.cache.CacheService;
+import org.gingesnap.cdc.configuration.Backend;
+import org.gingesnap.cdc.configuration.Region;
+import org.gingesnap.cdc.translation.ColumnJsonTranslator;
+import org.gingesnap.cdc.translation.ColumnStringTranslator;
+import org.gingesnap.cdc.translation.IdentityTranslator;
+import org.gingesnap.cdc.translation.JsonTranslator;
+import org.gingesnap.cdc.translation.PrependJsonTranslator;
+import org.gingesnap.cdc.translation.PrependStringTranslator;
+
+import io.quarkus.arc.Arc;
+
+public class EventProcessingChainFactory {
+
+   public static EventProcessingChain create(String name, Region.SingleRegion region) {
+      return EventProcessingChain.chained(
+            // The filter link should always be the head.
+            new EventFilterLink(region),
+
+            // The cache link should always be last.
+            new CacheBackendLink(createCacheBackend(name, region.backend()))
+      );
+   }
+
+   private static CacheBackend createCacheBackend(String name, Backend backend) {
+      JsonTranslator<?> keyTranslator;
+      JsonTranslator<?> valueTranslator = backend.columns().isPresent() ?
+            new ColumnJsonTranslator(backend.columns().get()) : IdentityTranslator.getInstance();
+      Optional<List<String>> optionalKeys = backend.keyColumns();
+      // TODO: hardcoded value here
+      List<String> columnsToUse = optionalKeys.orElse(List.of("id"));
+      switch (backend.keyType()) {
+         case PLAIN:
+            ColumnStringTranslator stringTranslator = new ColumnStringTranslator(columnsToUse, backend.plainSeparator());
+            keyTranslator = backend.prefixRuleName() ? new PrependStringTranslator(stringTranslator, name) : stringTranslator;
+            break;
+         case JSON:
+            ColumnJsonTranslator jsonTranslator = new ColumnJsonTranslator(columnsToUse);
+            // TODO: hardcoded value here
+            keyTranslator = backend.prefixRuleName() ? new PrependJsonTranslator(jsonTranslator, backend.jsonRuleName(), name) : jsonTranslator;
+            break;
+         default:
+            throw new IllegalArgumentException("Key type: " + backend.keyType() + " not supported!");
+      }
+
+      CacheService cacheService = Arc.container().instance(CacheService.class).get();
```

Comment:
Should we be using `Arc` to look this up here, or would it be better to pass the `CacheBackend` via `create` as it's a mandatory requirement?